### PR TITLE
apps/shell/tash_main.c: fix single quotes problem in ! command issue

### PR DESCRIPTION
**Pull request by Tirtha-Samsung**
```
TASH>>wm_test softap '@!123' '123123132' 1
TASH>>[WT][T18] wifi manager test!!     wm_test_main:947
[WT][T18] wait func signal      wm_test_main:958
[WT][T19] -->   _wt_softap_start:312
====================================
AP INFO
------------------------------------
SSID: @!123
channel: 1
====================================
[WM][INFO] --> wifi_manager_set_mode 136
[WM][INFO] handle request state(UNINITIALIZED) evt(CMD_SET_SOFTAP)
[WT][ERR][T19] Run SoftAP Fail  _wt_softap_start:331
[WT][T19] <--   _wt_softap_start:333
[WT][T19] send func signal      _wt_process:938

TASH>>
```

### CODE REVIEW

File: apps/shell/tash_main.c [hunk 1]
```diff
@@ -423,6 +424,13 @@ int tash_execute_cmdline(char *buff)
 					*(buff + 1) = ASCII_NUL;
 					break;
 
+				case ASCII_RSQUOTE:
+					/* Argument is started with opening single quotation mark */
```
Comment (abhishek-samsung) [inline apps/shell/tash_main.c:428]:
```suggestion
					/* Argument is starting with opening single quotation mark */
```

Comment (sunghan-chang) [reply apps/shell/tash_main.c:428]: @Tirtha-Samsung Could you check Abhishek's comment above?

File: apps/shell/tash_main.c [hunk 2]
```diff
@@ -423,8 +424,15 @@ int tash_execute_cmdline(char *buff)
 					*(buff + 1) = ASCII_NUL;
 					break;
 
+				case ASCII_RSQUOTE:
```
Comment (ewoodev) [inline apps/shell/tash_main.c:427]: How about below for consistency with `IN_APOSTROPHY` , `case IN_APOSTROPHY:`
`case ASCII_APOSTROPHY:`

File: apps/shell/tash_main.c [hunk 3]
```diff
@@ -37,7 +37,8 @@
 enum tash_input_state_e {
 	IN_VOID,
 	IN_WORD,
-	IN_QUOTE
+	IN_QUOTE,
+	IN_APOSTROPHE
```
Comment (sunghan-chang) [inline apps/shell/tash_main.c:41]: As I left before, we can add a new state here for `!`. I mean we can merge the functions for ! and others. Please consider this with new PR.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**